dspace-api/src/main/java/org/dspace/authenticate/ShibAuthentication.java

```diff
@@ -279,13 +279,6 @@ public int authenticate(Context context, String username, String password,
 	public int[] getSpecialGroups(Context context, HttpServletRequest request)
 	{
 		try {
-			// User has not successfuly authenticated via shibboleth.
-			if ( request == null || 
-					context.getCurrentUser() == null || 
-					request.getSession().getAttribute("shib.authenticated") == null ) {
-				return new int[0];
-			}
```

Comment:
Now I figured out, what is wrong: request.getSession().getAttribute("shib.authenticated") is null (even though I did authenticate via Shib) and this prevents the complete method from getting executed. I understand, that this check is important and would agree not to remove it, but why is the attribute not set properly here?